examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavigationActivity.kt

```diff
@@ -0,0 +1,303 @@
+                )
+            }
+            true
+        }
+        locationComponent = mapboxMap.locationComponent
+    }
+
+    @SuppressLint("RestrictedApi")
+    fun initLocationComponent(loadedMapStyle: Style, mapboxMap: MapboxMap) {
+        when (PermissionsManager.areLocationPermissionsGranted(this)) {
+            true -> {
+                mapboxMap.moveCamera(CameraUpdateFactory.zoomTo(15.0))
+                mapboxMap.locationComponent.let { locationComponent ->
+                    val locationComponentActivationOptions =
+                            LocationComponentActivationOptions.builder(this, loadedMapStyle)
+                                    .build()
+
+                    locationComponent.activateLocationComponent(locationComponentActivationOptions)
+                    locationComponent.isLocationComponentEnabled = true
+                    locationComponent.cameraMode = CameraMode.TRACKING
+                    locationComponent.renderMode = RenderMode.COMPASS
+
+                    initLocationEngine()
+                }
+            }
+            false -> {
+                permissionsManager =
+                        PermissionsManager(this).also { it.requestLocationPermissions(this) }
+            }
+        }
+    }
+
+    fun initLocationEngine() {
+        val requestLocationUpdateRequest =
+                LocationEngineRequest.Builder(DEFAULT_INTERVAL_IN_MILLISECONDS)
+                        .setPriority(LocationEngineRequest.PRIORITY_NO_POWER)
+                        .setMaxWaitTime(DEFAULT_MAX_WAIT_TIME)
+                        .build()
+
+        mapboxNavigation?.locationEngine?.requestLocationUpdates(
+                requestLocationUpdateRequest,
+                locationListenerCallback,
+                mainLooper
+        )
+        mapboxNavigation?.locationEngine?.getLastLocation(locationListenerCallback)
+    }
+
+    private val routesReqCallback = object : RoutesRequestCallback {
+        override fun onRoutesReady(routes: List<DirectionsRoute>) {
+            if (routes.isNotEmpty()) {
+                navigationMapboxMap?.drawRoute(routes[0])
+                if (shouldSimulateRoute()) {
+                    (mapboxNavigation?.locationEngine as ReplayRouteLocationEngine).assign(routes[0])
+                }
+                startNavigation.visibility = View.VISIBLE
+            } else {
+                startNavigation.visibility = View.GONE
+            }
+        }
+
+        override fun onRoutesRequestFailure(throwable: Throwable, routeOptions: RouteOptions) {
+            Timber.e("route request failure %s", throwable.toString())
+        }
+
+        override fun onRoutesRequestCanceled(routeOptions: RouteOptions) {
+            Timber.d("route request canceled")
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    fun initListeners() {
+        startNavigation.setOnClickListener {
+            if (mapboxNavigation?.getRoutes()?.isNotEmpty() == true) {
```

Comment:
`== true` typo?

Reply:
Because the entire statement uses null safety '?' in the statement the compiler complains if there's no concrete value to compare against.